crates/bevy_app/src/app.rs

```diff
@@ -203,85 +201,89 @@ impl App {
-    }
-
-    pub fn add_system_to_stage<Params>(
-        &mut self,
-        stage_label: impl StageLabel,
-        system: impl IntoSystemDescriptor<Params>,
-    ) -> &mut Self {
-        self.schedule.add_system_to_stage(stage_label, system);
-        self
-    }
-
-    pub fn add_system_set_to_stage(
-        &mut self,
-        stage_label: impl StageLabel,
-        system_set: SystemSet,
-    ) -> &mut Self {
-        self.schedule
-            .add_system_set_to_stage(stage_label, system_set);
+    pub fn add_system<Params>(&mut self, system: impl IntoSystem<(), (), Params>) -> &mut Self {
+        let system = system.system();
+        if system.config().startup {
+            let system = {
+                if system.config().stage.is_some() {
+                    system
+                } else {
+                    system.stage(StartupStage::Startup)
+                }
+            };
+            self.schedule
+                .stage(CoreStage::Startup, |schedule: &mut Schedule| {
+                    schedule.add_system(system)
+                });
+        } else {
+            let system = {
+                if system.config().stage.is_some() {
+                    system
+                } else {
+                    system.stage(CoreStage::Update)
+                }
+            };
+            self.schedule.add_system(system);
+        }
         self
     }
 
-    /// Adds a system that is run once at application startup
-    ///
-    /// Startup systems run exactly once BEFORE all other systems. These are generally used for
-    /// app initialization code (ex: adding entities and resources).
-    ///
-    /// * For adding a system that runs for every frame, see [`App::add_system`].
-    /// * For adding a system to specific stage, see [`App::add_system_to_stage`].
-    ///
-    /// ## Example
-    /// ```
-    /// # use bevy_app::prelude::*;
-    /// # use bevy_ecs::prelude::*;
-    /// #
-    /// fn my_startup_system(_commands: Commands) {
-    ///     println!("My startup system");
-    /// }
-    ///
-    /// App::new()
-    ///     .add_startup_system(my_startup_system);
-    /// ```
-    pub fn add_startup_system<Params>(
-        &mut self,
-        system: impl IntoSystemDescriptor<Params>,
-    ) -> &mut Self {
-        self.add_startup_system_to_stage(StartupStage::Startup, system)
-    }
-
-    pub fn add_startup_system_set(&mut self, system_set: SystemSet) -> &mut Self {
-        self.add_startup_system_set_to_stage(StartupStage::Startup, system_set)
-    }
-
-    pub fn add_startup_system_to_stage<Params>(
-        &mut self,
-        stage_label: impl StageLabel,
-        system: impl IntoSystemDescriptor<Params>,
-    ) -> &mut Self {
-        self.schedule
-            .stage(CoreStage::Startup, |schedule: &mut Schedule| {
-                schedule.add_system_to_stage(stage_label, system)
-            });
+    pub fn add_system_set(&mut self, set: SystemSet) -> &mut Self {
```

Comment:
I think there should be some kind of validation here if a system inside the system set has a conflicting stage.